src/seedsigner/views/settings_views.py

```diff
@@ -178,7 +180,8 @@ def run(self):
 
         # All selects stay in place; re-initialize where in the list we left off
         self.selected_button = ret_value
-        return self.run()
+
+        return Destination(SettingsEntryUpdateSelectionView, view_args=dict(attr_name=self.settings_entry.attr_name, parent_initial_scroll=self.parent_initial_scroll, selected_button=self.selected_button), skip_current_view=True)
```

Comment:
todo: delete this comment once understood why this has changed so much.

Reply:
@jdlcdl The recursive `self.run()` made it impossible for the flow-based tests to track the click-to-click progression through the settings update. So instead we just re-call the same View with the same params to achieve the same end result (re-display the associated Screen and await next user input).